src/prototype_room_my.js

```diff
@@ -397,6 +339,66 @@ Room.prototype.executeRoom = function() {
       Game.notify(this.name + ' Under attack from ' + hostiles[0].owner.username + ' at ' + Game.time);
     }
   }
+};
+
+Room.prototype.executeRoom = function() {
+  let cpuUsed = Game.cpu.getUsed();
+  this.buildBase();
+  this.memory.attackTimer = this.memory.attackTimer || 0;
+
+  var spawns = this.find(FIND_MY_STRUCTURES, {
+    filter: function(object) {
+      return object.structureType === STRUCTURE_SPAWN;
+    }
+  });
+
+  var hostiles = this.find(FIND_HOSTILE_CREEPS, {
+    filter: this.findAttackCreeps
```

Comment:
hum dunno, don't have friends xD but i didn't edit that... A good way for always filter them could be to do a general filter method in each room for drop different structure / creeps in differents array as we no more need to filter them later. i'll think about